src/shims/unix/fd.rs

```diff
@@ -329,7 +313,8 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         let fd = this.read_scalar(fd_op)?.to_i32()?;
 
         Ok(Scalar::from_i32(if let Some(file_descriptor) = this.machine.fds.remove(fd) {
-            let result = file_descriptor.close(this.machine.communicate())?;
+            let result =
+                file_descriptor.0.as_ref().borrow_mut().close(this.machine.communicate())?;
```

Comment:
so.. we'll want to check the refcount here first, and only call close if it's 1

Reply:
refactored the close check to another function, thanks for the review :)